db/seeds.d/101-locations.rb

```diff
@@ -4,9 +4,15 @@
 # !!! PLEASE KEEP THIS SCRIPT IDEMPOTENT !!!
 #
 
-if Location.exists? && !Location.default_location
+if Location.exists? &&
+    !Setting[:default_location_subscribed_hosts].present? ||
+    !Setting[:default_location_puppet_content].present?
   # Create a new location to be used as the Katello Default.
-  Location.create!(:name => "Default Location") do |loc|
-    loc.katello_default = true
+  default_location = Location.where(:name => ENV['SEED_LOCATION']).first_or_create
```

Comment:
@dLobatog how this is supposed to work if I don't set SEED_LOCATION env variable? first_or_create tries to create a new location with nil as a name which fails silently

Reply:
In katello there is always a SEED_LOCATION at install time, and after the initial run, the two default_location Settings get set below.  At a 2nd run (where the env variable is not set), the if check above checking those settings should fail and this whole block is skipped).  

I guess there is nothing stopping a user from clearing out those settings manually (although pieces of katello likely wouldn't function properly).   A more explicit check would probably be good :)

Reply:
you mean at install time when installer is used? so people who do git clone are not able to seed without finding out the right env variable to set, would it be acceptable to set a default value here? if installer always set it, it should do no harm... I'm happy to send a PR if you agree

Reply:
yeah, i was more referring to production environments.  I agree this should be handled more safely for manual git-based setups.